src/coreComponents/physicsSolvers/inducedSeismicity/SeismicityRateBase.cpp

```diff
@@ -0,0 +1,257 @@
+      subRegion.registerField< inducedSeismicity::meanShearStress_n >( getName() );
+
+      subRegion.registerField< inducedSeismicity::seismicityRate >( getName() );
+    } );
+   } );
+}
+
+void SeismicityRateBase::initializePreSubGroups()
+{
+  SolverBase::initializePreSubGroups();
+
+  DomainPartition & domain = this->getGroupByPath< DomainPartition >( "/Problem/domain" );
+  forDiscretizationOnMeshTargets( domain.getMeshBodies(), [&]( string const &,
+                                                               MeshLevel & mesh,
+                                                               arrayView1d< string const > const & regionNames )
+
+  {
+    mesh.getElemManager().forElementSubRegions( regionNames,
+                                                [&]( localIndex const,
+                                                     ElementSubRegionBase & subRegion )
+    {
+      arrayView1d< real64 > const tempR = subRegion.getField< inducedSeismicity::seismicityRate >();
+      tempR.setValues< parallelHostPolicy >( 1.0 );
+    } );
+  } );
+}
+
+void SeismicityRateBase::updateMeanSolidStress( ElementSubRegionBase & subRegion )
+{
+  // Retrieve field variables
+  arrayView2d< real64 > const meanStress = subRegion.getField< inducedSeismicity::meanStress >();
+  meanStress.setValues< parallelHostPolicy >( 0.0 );
+
+  arrayView1d< real64 > const sig = subRegion.getField< inducedSeismicity::meanNormalStress >();
+  arrayView1d< real64 > const sig_n = subRegion.getField< inducedSeismicity::meanNormalStress_n >();
+  
+  arrayView1d< real64 > const tau = subRegion.getField< inducedSeismicity::meanShearStress >();
+  arrayView1d< real64 > const tau_n = subRegion.getField< inducedSeismicity::meanShearStress_n >();
+
+  // Retrieve stress state computed by m_stressSolver, called in solverStep
+  string const & solidModelName = subRegion.getReference< string >( SolidMechanicsLagrangianFEM::viewKeyStruct::solidMaterialNamesString());
+  SolidBase const & solidModel = getConstitutiveModel< SolidBase >( subRegion, solidModelName );
+  arrayView3d< real64 const, solid::STRESS_USD > const stress = solidModel.getStress();
+  
+  // loop over all elements and computed unweighted average across all nodes 
+  // as average stress state acting over the cell ceneter
+  // TODO: APPLY WEIGHTS TO AVERAGE TO ACCOUNT FOR ELEMENT SHAPE
+  forAll< parallelDevicePolicy<> >(  sig.size(), [=] GEOS_HOST_DEVICE ( localIndex const k )
+  {
+    // update projected stresses from previou step
+    sig_n[k] = sig[k];
+    tau_n[k] = tau[k];
+
+    // Compute average stress state
+    for( int q = 0; q < stress.size(1); q++ ) 
+    {
+      LvArray::tensorOps::add< 6 >( meanStress[k], stress[k][q] );
+    }
+    LvArray::tensorOps::scale< 6 >(meanStress[k], 1./stress.size(1));
+
+    // Project average stress state to fault
+    sig[k] = LvArray::tensorOps::AiBi< 6 >( meanStress[k], m_faultNormalVoigt);
```

Comment:
This is great if the user can provide a desired fault orientation.  Is it useful to add the option to find the "critical orientation"--that is, if the medium contained fractures in all direction, which orientation is most likely to be activated?  This would be a lower bound.

Reply:
I think this is a good idea. It would require solving for the optimal orientation given the current stress tensor. We could mark it as an issue.